components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/engine/runtime/xpath/JaxpFunctionResolver.java

```diff
@@ -590,11 +724,695 @@ private Node createOrgEntity(GenericHumanRoleDAO ghr) {
+     */
+    public class GetCountOfSubTasksInState implements XPathFunction {
+
+        @Override
+        public Object evaluate(List args) throws XPathFunctionException {
+            {
+                if (args.size() == 1) {
+                    //Case 1 : Consider current task
+                    return countSubTasks(args.get(0));
+                } else if ((args.size() == 2) && (args.get(1) instanceof String)) {
+                    String taskName = (String) args.get(1);
+                    if (evalCtx.getTask().getName().equals(taskName)) {
+                        //Case 2 : TaskName equals to current task, consider current task
+                        return countSubTasks(args.get(0));
+                    } else {
+                        // Case 3: Getting specific task. Unsupported.
+                        log.warn("HumanTask Xpath: getCountOfSubTasks(\"" + taskName + "\")"
+                                + " operation is not currently supported in this version of WSO2 BPS."
+                                + " Use getCountOfSubTasks() instead.");
+                    }
+                } else {
+                    throw new HumanTaskRuntimeException("Invalid number of arguments : " + args.size()
+                            + ", for function getCountOfSubTasksInState()");
+                }
+                return 0;
+            }
+        }
+
+        private Integer countSubTasks(Object arg) {
+            int count = 0;
+            if (arg instanceof String && isValidStatus((String) arg)) {
+                //correct argument
+                List<TaskDAO> subTaskList = evalCtx.getTask().getSubTasks();
+                String state = (String) arg;
+                for (TaskDAO subTask : subTaskList) {
+                    if (state.equalsIgnoreCase(subTask.getStatus().name())) {
+                        count++;
+                    }
+                }
+            } else {
+                //invalid argument
+                throw new HumanTaskRuntimeException(
+                        "Invalid argument : " + arg + ", for function getCountOfSubTasksInState()");
+            }
+
+            return count;
+        }
+    }
+
+    //check the validity of a given task status
+    private boolean isValidStatus(String status) {
+        for (TaskStatus taskStatus : TaskStatus.values()) {
+            if (status.equalsIgnoreCase(taskStatus.name())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns the concatenation of all string nodes,
+     * returns an empty string for an empty node-set
+     * logic: goes through the list and concat textContent to result string,
+     * return empty string for empty list.
+     * In : node-set of string nodes
+     */
+    public class Concat implements XPathFunction {
+
+        @Override
+        public Object evaluate(List args) throws XPathFunctionException {
+            String result = "";
```

Comment:
According to the spec [1], we should return empty string for empty node sets. That is the reason to use empty result. In that case this is correct I guess. Thank You.
[1] - http://docs.oasis-open.org/bpel4people/ws-humantask-1.1-spec-cs-01.html